pip/qsharp/_qsharp.py

```diff
@@ -1,10 +1,33 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT License.
 
-from ._native import Interpreter
+from ._native import Interpreter, TargetProfile
 
-# Create a Q# interpreter singleton.
-_interpreter = Interpreter()
+_interpreter = None
+
+
+def init(target_profile: TargetProfile = TargetProfile.Full) -> None:
+    """
+    Initializes the Q# interpreter.
+
+    :param target_profile: Setting the target profile allows the Q#
+        interpreter to generate programs that are compatible
+        with a specific target. See :py:class: `qsharp.TargetProfile`.
+    """
+    global _interpreter
+    _interpreter = Interpreter(target_profile)
+
+
+def get_interpreter() -> Interpreter:
+    """
+    Returns the Q# interpreter.
+
+    :returns: The Q# interpreter.
+    """
+    global _interpreter
+    if _interpreter is None:
+        init()
```

Comment:
@minestarks I changed this to auto-init, so it's not required for folks who just want the default, non-hardware compatible settings. If we want to go further and force init to be called only once, we could have `init` raise an error if `_interpreter` is not `None`, which would force the kernel restart.